#!/usr/bin/env python
PACKAGE = "hippo_chain"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("kSigma1",  double_t,   0,  "Linear error constant",            0.1,    0.0,    10.0)
gen.add("kSigma2",  double_t,   0,  "Angular error constant",           0.1,    0.0,    10.0)
gen.add("kP",       double_t,   0,  "Proportional controller gain",     0.1,    0.0,    10.0)
gen.add("kSat",     double_t,   0,  "Gain of saturated part",           0.1,    0.0,    10.0)
gen.add("lim",      double_t,   0,  "Saturation limit",                 0.1,    0.0,    10.0)
gen.add("update",   bool_t,     0,  "Update Parameters",                True)

exit(gen.generate(PACKAGE, "hippo_chain_node", "BaseController"))
